dumper/app/src/test/java/com/google/edwmigration/dumper/application/dumper/MetadataDumperTest.java

```diff
@@ -25,10 +33,123 @@
 @RunWith(JUnit4.class)
 public class MetadataDumperTest {
 
+  private File file;
+  private MetadataDumper dumper = new MetadataDumper().withExitOnError(false);
+  Connector connector = new HiveMetadataConnector();
+
+  @After
+  public void tearDown() throws IOException {
+    if (file == null || !file.exists()) return;
+    if (file.isDirectory()) FileUtils.deleteDirectory(file);
+    else file.delete();
+  }
+
   @Test
   public void testInstantiate() throws Exception {
-    MetadataDumper dumper = new MetadataDumper();
-    // dumper.run("--help");    // Calls System.exit(1)
+    dumper = new MetadataDumper();
     dumper.run("--connector", new BigQueryLogsConnector().getName(), "--dry-run");
   }
+
+  @Test
+  public void testCreatesDefaultOutputZip() throws Exception {
+    // Arrange
+    String name = connector.getDefaultFileName(false);
+    file = new File(name);
+
+    // Act
+    dumper.run("--connector", connector.getName());
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testCreatesDefaultOutputZipInProvidedDirectory() throws Exception {
+    // Arrange
+    String name = connector.getDefaultFileName(false);
+    String path = "dir";
+    file = new File(path, name);
+
+    // Act
+    dumper.run("--connector", connector.getName(), "--output", path);
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testCreatesZipWithGivenName() throws Exception {
+    // Arrange
+    String name = "test.zip";
+    file = new File(name);
+
+    // Act
+    dumper.run("--connector", connector.getName(), "--output", name);
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testOverridesZipWithGivenName() throws Exception {
+    // Arrange
+    String name = "test-dir/test.zip";
+    file = new File(name);
+
+    // Act
+    Files.createParentDirs(file);
+    file.createNewFile();
+    dumper.run("--connector", connector.getName(), "--output", name);
+
+    // Assert
+    Assert.assertTrue(file.exists());
+  }
+
+  @Test
+  public void testOverridesZipWithDefaultName() throws Exception {
+    // Arrange
+    File expectedFile = new File("test-dir/dwh-migration-hiveql-metadata.zip");
+    String name = "test-dir";
+    file = new File(name);
+
+    // Act
+    file.mkdirs();
+    dumper.run("--connector", connector.getName(), "--output", name);
+
+    // Assert
+    Assert.assertTrue(expectedFile.exists());
+  }
+
+  @Test
+  public void testCreatesFileInsideFolderNameWithZip() throws Exception {
+    // Arrange
+    File expectedFile = new File("dir.zip/dwh-migration-hiveql-metadata.zip");
+    String name = "dir.zip";
+    file = new File(name);
+
+    // Act
+    file.mkdirs();
```

Comment:
Optional: No need to change anything. This test case is fine. Another interesting one would be the same test, but without `mkdirs`, because the directory should be created automatically:

https://github.com/google/dwh-migration-tools/blob/82a32fa00e380294dc8d50d36d165909ef9423f5/dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/MetadataDumper.java#L262

Reply:
Resolved in chat